### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by pbignardi**
Addresses issue #23 

Outline of the changes:

### Extracted colors from ```editor.css``` into separate css files
All the colors in ```editor.css``` have been replaced by CSS variables that are defined in ```light_color.css``` and ```dark_color.css```.
These two files contains media queries to decide which of the variables to use depending on the prefered user theme.
For simplicity, all the variables are set in the ```:root``` scope, but the names are quite descriptive, in case they need to be modified down the line.

### Included ```cm-cursor``` and ```cm-docs-binding``` styling in ```editor.css```
Styling for ```cm-cursor``` and ```cm-docs-binding``` classes are moved in ```editor.css``` for clarity and improve code maintainability. CSS variables are created in ```light_color.css``` and ```dark_color.css``` to specify the colors.

### Modified ```CellInput.js``` and ```FilePicker.js``` to allow for dark theming
The default code-mirror setup only support light theming. An optional parameter was added to allow for dark theming in the cell input and file picker.

### SVG graphics inverted
SVG graphics are reused with appropriate filters in ```editor.css```. A good improvement would be to __create custom svg graphics for the dark theme__.

# Try this Pull Request!
Open Julia and type:
```jl
julia> ]
(@v1.6) pkg> activate --temp
(@v1.6) pkg> add https://github.com/pbignardi/Pluto.jl#main
(@v1.6) pkg> # press backspace/delete to go back
julia> import Pluto; Pluto.run()
```

### CODE REVIEW

**Comment by fonsp** [comment]
This looks really good @pbignardi, thanks so much!

**Comment by pankgeorg** [comment]
This is awesome, from a software engineering perspective! I tried to push a commit with some styling adjustments - can you give us access? I'll get a more detailed list soon!

Great work overall!!

**Comment by pbignardi** [comment]
Thank you for the kind words 😄 Shall I set you as collaborator in my forked repo?

**Comment by pbignardi** [comment]
I also pushed some changes to resolve the conflicts earlier this morning.

**Comment by pankgeorg** [comment]
> Thank you for the kind words 😄 Shall I set you as collaborator in my forked repo?

Can you check if you can do [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) process? if yes, it will be easier

**Comment by pbignardi** [comment]
> > Thank you for the kind words 😄 Shall I set you as collaborator in my forked repo?
> 
> Can you check if you can do [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) process? if yes, it will be easier

Done, box checked. Sorry, it's my first PR 😅

**Comment by pankgeorg** [comment]
Oops I closed this by mistake

**Comment by pbignardi** [comment]
Sorry, I'm quite newbie in project contributing. Are there any changes that I need to do to the code?

**Comment by fonsp** [comment]
Hi @pbignardi, thanks again for contributing! I think there were some technical problems because this PR is coming from the `main` branch of your fork instead of a [new branch for your PR](https://jarv.is/notes/how-to-pull-request-fork-github/). Something to remember for your next contribution to Pluto!

I think the PR will take a couple of weeks of tweaking from our side until we are happy with how all the styles look, and we will have to try how well it plays together with other packages, but the technical side looks really good!

**Comment by pbignardi** [comment]
Oh, ok thank you for the heads up! I'll start looking for a new issue to contribute to 😄

File: frontend/components/CellInput.js [hunk 1]
```diff
@@ -53,33 +53,50 @@ import { cell_movement_plugin, prevent_holding_a_key_from_doing_things_across_ce
...
+    link,
+    invalid,
+    invalid_bg } from "./SyntaxColorsParser.js"
 
 export const pluto_syntax_colors = HighlightStyle.define([
     /* The following three need a specific version of the julia parser, will add that later (still messing with it 😈) */
     // Symbol
-    { tag: tags.literal, color: "#5e7ad3", fontWeight: 700 },
-    { tag: tags.macroName, color: "#5668a4", fontWeight: 700 },
+    { tag: tags.literal, color: literal, fontWeight: 700 },
```
Comment (fonsp) [inline frontend/components/CellInput.js:77]:
```suggestion
    { tag: tags.literal, color: "var(--cm-builtin-color)", fontWeight: 700 },
```

It turns out you can use CSS variables here! The would simplify our code, and it means that you don't need to reload Pluto after switching to dark mode 🌝

**Review by fonsp — CHANGES_REQUESTED** [review]

**Comment by fonsp** [comment]
Sorry for the slow response! The reason is: we think dark mode is a really exciting feature, and we are saving it up for our next big release, either `0.18.0` (or `1.0.0` if that is soon). 🙊

File: frontend/components/CellInput.js (hunk 1 above)
Comment (pbignardi) [reply frontend/components/CellInput.js:77]: Good idea! So basically for all the variables imported from `SyntaxColorParser.js` are not needed, right? You would only refer to the CSS variable directly. Am I understanding this correctly?

**Comment by pbignardi** [comment]
It's exciting to see so much hype for this feature 😄 😎

**Comment by pbignardi** [comment]
Just noticed the background in the dark mode. Very nice, I like it! 👍🏻

**Comment by fonsp** [comment]
TODO:
- [ ] errored focused cell in light mode has glitched traffic light
- [ ] selection highlight color in CM needs more contrast

Future PR:
- Pattern for background on main menu and large screens!

**Comment by icweaver** [comment]
You got thiss
teddy-bear-break-dancing

**Comment by pbignardi** [comment]
> * [ ]  errored focused cell in light mode has glitched traffic light

@fonsp Are you referring to this?
Screenshot 2022-01-12 at 22 35 29

**Comment by pbignardi** [comment]
> * [ ]  selection highlight color in CM needs more contrast

Regarding this problem, the best solution would be to modify the CodeMirror config files, since in there it is defined the selection background color that it's used when in dark mode. Alternatively we could also bodge it by forcing such color afterwards by changing it in the css and using the _!important_ keyword; not the cleanest solution but it works.

**Comment by pankgeorg** [comment]
> > * [ ]   selection highlight color in CM needs more contrast
> 
> Regarding this problem, the best solution would be to modify the CodeMirror config files, since in there it is defined the selection background color that it's used when in dark mode. Alternatively we could also bodge it by forcing such color afterwards by changing it in the css and using the _!important_ keyword; not the cleanest solution but it works.
> 
> Edit: fyi, in the `index.es.js` file the selection background color is at line 10862.

there is a clean CSS solution, but we are looking for the right candidate color!!

**Comment by pbignardi** [comment]
Fair point 😄

**Comment by pbignardi** [comment]
Ok I updated the color, I made it green. See if you like it 😉

**Comment by diegozea** [comment]
This is awesome! :tada: I want to test it; how should we do to activate the dark mode? :new_moon:

**Comment by pankgeorg** [comment]
Just update to the latest version and set you operating system theme to dark!

**Comment by diegozea** [comment]
It seems that it doesn't work in Ubuntu 18.04. I have followed these instructions to set the dark mode on the system: https://nixytrix.com/how-do-i-enable-dark-mode-in-ubuntu-18-04/

**Comment by pankgeorg** [comment]
What pluto version are you on?

**Comment by diegozea** [comment]
No, it doesn't work with the GitHub page either. So, it should be a problem with `Ubuntu 18.04` and `Tweaks`.

**Pull request merged**